test/Test/Util.hs

```diff
@@ -0,0 +1,16 @@
+{-# LANGUAGE LinearTypes #-}
+
+module Test.Util where
+
+import Data.Unrestricted.Linear
+import qualified Prelude.Linear as Lin
+
+-- XXX: This should be in linear base
+-- This is also a terrible name
+getSnd :: (Consumable a, Movable b) => (a, b) #-> Unrestricted b
+getSnd (a, b) = lseq a (move b)
```

Comment:
This is too bespoke to be anywhere. The main reason why you need it is because the `read` functions currently omits to return an `Unrestricted` value. Also because we are lacking proper support for linear cases, I'm afraid, but that's not on you.